core/src/main/java/org/elasticsearch/common/io/stream/Writeable.java

```diff
@@ -31,6 +31,11 @@
  *
  * Prefer implementing this interface over implementing {@link Streamable} where possible. Lots of code depends on {@linkplain Streamable}
  * so this isn't always possible.
+ *
+ * The fact that this interface extends {@link StreamableReader} should be consider vestigial. Instead of using its
```

Comment:
do we want to add a TODO to mention that we are going to remove readFrom if that is the direction?